www/modules/centreon-open-tickets/providers/GlpiRestApi/GlpiRestApiProvider.class.php

```diff
@@ -1076,21 +1088,55 @@ protected function curlQuery($info) {
             }
         }
 
+        // parse headers to manage pagination
+        if ($offset !== null) {
+            $curlHeaders = [];
+            curl_setopt($curl, CURLOPT_HEADERFUNCTION, function ($curlResource, $curlHeader) use (&$curlHeaders) {
+                $length = strlen($curlHeader);
+                $curlHeader = explode(':', $curlHeader, 2);
+
+                if (count($curlHeader) < 2) {
+                    return $length;
+                }
+
+                $curlHeaders[strtolower(trim($curlHeader[0]))][] = trim($curlHeader[1]);
+
+                return $length;
+            });
+        }
+
         // execute curl and get status information
         $curlResult = json_decode(curl_exec($curl), true);
         $httpCode = curl_getinfo($curl, CURLINFO_HTTP_CODE);
         curl_close($curl);
 
-        // if http is 401 and message is about token, perhaps the token has expired, so we get a new one
-        if ($httpCode == 401 && $curlResult[0] == 'ERROR_SESSION_TOKEN_INVALID') {
+        if ($httpCode === 206 && $offset !== null) {
+            // If partial content, get next page
+            if (preg_match('/\/(\d+)/', $curlHeaders['content-range'][0], $matches)) {
+                $total = $matches[1];
+
+                $offset = $offset + self::PAGE_SIZE;
+
+                if ($offset <= $total) {
+                    $curlResult = array_merge_recursive(
+                        $curlResult,
+                        $this->curlQuery($info, $offset)
+                    );
+                }
+            }
+        } elseif ($httpCode == 401 && $curlResult[0] == 'ERROR_SESSION_TOKEN_INVALID') {
+            // if http is 401 and message is about token, perhaps the token has expired, so we get a new one
             try {
                 $this->initSession();
-                $this->curlQuery($info);
+                $this->curlQuery($info, $offset);
             } catch (\Exception $e) {
                 throw new \Exception($e->getMessage(), $e->getCode());
             }
-        // for any other issue, we throw an exception
+        } elseif ($httpCode === 206) {
+            throw new Exception('ENDPOINT: ' . $apiAddress . ' || GLPI ERROR : ' . $curlResult[0] .
+            ' || GLPI MESSAGE: ' . $curlResult[1] . ' || HTTP ERROR: ' . $httpCode, 11);
         } elseif ($httpCode >= 400) {
+            // for any other issue, we throw an exception
             throw new Exception('ENDPOINT: ' . $apiAddress . ' || GLPI ERROR : ' . $curlResult[0] .
             ' || GLPI MESSAGE: ' . $curlResult[1] . ' || HTTP ERROR: ' . $httpCode, 11);
```

Comment:
```suggestion
        } elseif ($httpCode === 206 || $httpCode >= 400) {
            throw new Exception('ENDPOINT: ' . $apiAddress . ' || GLPI ERROR : ' . $curlResult[0] .
            ' || GLPI MESSAGE: ' . $curlResult[1] . ' || HTTP ERROR: ' . $httpCode, 11);
        }
```

Reply:
the first elseif was useless, I removed it